eventuals/concurrent.h

```diff
@@ -285,12 +286,15 @@ struct _Concurrent final {
 
                 k.Start(); // Exits the synchronized block!
               })
-              .fail([this, fiber](auto& k, auto&& error) {
+              .fail([this, fiber, &exception](auto& k, auto&& error) {
                 fiber->done = true;
 
-                if (!exception_) {
-                  exception_ = make_exception_ptr_or_forward(
-                      std::forward<decltype(error)>(error));
+                if (!exception) {
+                  // Can't provide compile-time guarantees that it won't be executed at runtime, so
+                  // have to check explicitly.
+                  if constexpr (!std::is_same_v<std::decay_t<decltype(exception.value())>, Stopped>) {
+                    exception.emplace(std::forward<decltype(error)>(error));
+                  }
```

Comment:
I'm not following why you need to do this `if constexpr` here or what the comment is trying to say?